### ISSUE TITLE
Unable to create .log file in /mnt/c and also unexpected behavior when using .log with docker volumes

### ISSUE THREAD

**Issue by rafael-lua** [microsoft/WSL#7084]
### Windows Build Number

Microsoft Windows [versão 10.0.18362.1082]

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.4.72

### Distro Version

Ubuntu 20.04

### Other Software

Docker Desktop (Windows), version 3.4.0
Docker version 20.10.7, build f0df350

### Repro Steps

Currently, I can reproduce the error in two ways:

1. Trying to create text.log inside the /mnt/c/ with touch, vim etc...
```
cd /mnt/c/Users
touch text.log
```

2. Trying to mount volume with a text.log created inside the wsl file system to a docker container (While inside the wsl, I'm able to create a text.log file, but still cause issues)
```
cd ~
touch text.log
```

![logissue](https://user-images.githubusercontent.com/54607854/122566173-a1ac8880-d01d-11eb-914a-2d11ca1bfa26.PNG)

Also, related unanswered stackoverflow issue: https://stackoverflow.com/questions/63132041/wsl2-cannot-write-log-file 

### Expected Behavior

1. It should create a file text.log in the /mnt/c/Users like any other file.

2. It should be correctly mounted as a volume in the docker run command.

### Actual Behavior

1. Trying to create the text.log file in the /mnt/c/Users returns the following error: 
`touch: cannot touch 'text.log': No such file or directory`

2. Trying to mount the volume with a text.log file returns the following error: 
`docker: Error response from daemon: not a directory.`

If a text.log is not present when mounting the volume, docker will create a text.log DIRECTORY, which is the expected behavior from the -v command I believe. Mounting directories and other types of files works as well.

All the errors are shown in the repro steps image.

**Comment by therealkenc** [comment]
You (your Windows user) won't have access to `C:\Users` which is by design.

![image](https://user-images.githubusercontent.com/7049700/122619598-a6853280-d045-11eb-8608-29b19ddd828a.png)

I do not get the "`No such file or directory`" (I get `Permission denied`) but the underlying gist is the same. The `No such file or directory` difference in error code between our runs _might_ have been addressed since 18362.

If you _did_ give yourself (your Windows user) permission to `C:\Users` you can hypothetically list the steps you did to enable that, but the long-short is you want to be in `C:\Users\you` aka `/mnt/c/Users/you` not `C:\Users`, whatever the file (being a Docker scenario or otherwise). The repro is really the `touch log.txt`.

**Comment by rafael-lua** [comment]
Hm... I don't think I remember when and if I gave myself permission, I never use /mnt/c. When trying to solve my problem with the .log file in the docker volume, someone helping me asked me to try using the /mnt/c/Users, which is when I found out about the issue. The funny part is, the first time I did the mount on this directory, it worked, the touch worked and created a text.log. But the next try, it stopped working with this error.

Investigating more, I can also do a `touch anyothername.log` in the /c/mnt/Users, and it works.

It is the `touch text.log` that gives the error. 

If I try doing a `touch text.log` in in the /mnt/c/Users/Usuario, it WORKS as well. What?

Why only this specific file/path gives the issue? Is it possible that docker is doing something weird in some paths?

.....

I did restart my computer, also resetting docker... And got it back to normal, I think (touch text.log is working again). I have no idea what happened, but I'm pretty sure docker was the problem, and the volume thing. 

I'm sorry for creating a issue that ended being solved with a reset, but I was so lost on what the hell happened.
